### PR TITLE
Migrate VDT ITs to Wazuh repo

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
|#17403 |

## Description

This Issue aims to migrate the Vulnerability Detector IT suite from the QA repository.

## Tests
| path | Test Execution | OS |Commit | Notes |
|--|--|--|--|--|
| test_vulnerability_detector |  [:green_circle::green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh/files/12320991/VDT-Tier2-Results.zip) |  Local Manager | https://github.com/wazuh/wazuh/pull/17576/commits/acbde76609a3d9d12e0f4e345ca3078e2a8d18b6 | 4 Executions of migrated Tier2 VDT tests. Tier 0 and 1 are run through Github actions.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_cpe_helper/data/configuration_template/configuration_cpe_indexing.yaml
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_cpe_helper/data/configuration_template/configuration_cpe_indexing.yaml:1]: Folder name should be `configuration_templates`. Apply this change in all tests.

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection/index.html
+
+tags:
+    - vulnerability_detector
+    - cpe_helper
+'''
+import os
```
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py:47]: Remove this. Use Path:
```suggestion
from pathlib import Path
```
Apply this in all tests.

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+                                             configuration_metadata)
+systems = [metadata['system'] for metadata in configuration_metadata]
+
+# Set offline custom feeds configuration
+configurations = update_configuration_template(configurations, ['CUSTOM_NVD_JSON_PATH'], [custom_nvd_json_feed_path])
+
+agent_packages = [VULNERABLE_PACKAGES_EMPTY_VENDOR for metadata in configuration_metadata]
+
+
+@pytest.mark.tier(level=2)
```
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py:83]: Don't you think it is more organized to put this in pytestmark? If so, apply this in all tests.

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+from wazuh_testing.utils.configuration import (load_configuration_template, get_test_cases_data,
+                                        update_configuration_template)
+from wazuh_testing.utils.callbacks import generate_callback
+from wazuh_testing.modules.vulnerability_detector import patterns as cb
+from wazuh_testing.modules import vulnerability_detector as vd
+from wazuh_testing.modules.vulnerability_detector.packages import VULNERABLE_PACKAGES_EMPTY_VENDOR
+from wazuh_testing.tools.file_monitor import FileMonitor
+from . import TEST_CASES_PATH, CONFIGURATIONS_PATH, custom_nvd_json_feed_path
+
+local_internal_options = vd.LOCAL_INTERNAL_OPTIONS
```
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py:61]: Fix this after the changes in the framework here and in all tests.

File: tests/integration/test_vulnerability_detector/test_feeds/data/configuration_template/configuration_cpe_indexing.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      elements:
+        - disabled:
+            value: 'yes'
+
+    - section: wodle
+      attributes:
+        - name: 'syscollector'
+      elements:
+        - disabled:
+            value: 'yes'
```
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_feeds/data/configuration_template/configuration_cpe_indexing.yaml:75]: Missing new line at the end. Review this in all files of the PR.

File: tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+t1_configurations = update_configuration_template(t1_configurations, ['CUSTOM_NVD_JSON_PATH'],
+                                                  [custom_nvd_json_feed_path])
+t2_configurations = update_configuration_template(t2_configurations, ['CUSTOM_NVD_JSON_PATH', 'CUSTOM_MSU_JSON_PATH'],
+                                                  [custom_nvd_json_feed_path, custom_msu_feed_path])
+# Global vars
+agent_packages = VULNERABLE_PACKAGES
+
+
+@pytest.fixture()
+def prepare_scan(agent_system, mock_agent_with_custom_system):
```
Comment (TomasTurina) [inline tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py:95]: This has to be in a `conftest.py` file, not here. Apply this to all fixtures located in test files.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

**Comment by TomasTurina** [comment]
You need to rebase.

File: tests/integration/test_vulnerability_detector/conftest.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+        agent_system (str): System to set to the mocked agent.
+    """
+    prepare_scan_with_vuln_packages_implementation(agent_id=mock_agent_with_custom_system, last_scan=1,
+                                                   agent_system=agent_system)
+
+    yield mock_agent_with_custom_system
+
+
+@pytest.fixture()
+def prepare_full_scan_with_vuln_package(mock_agent_with_custom_system: int, agent_system: str) -> int:
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/conftest.py:114]: It is not duplicated. This one inserts only one package into the agent. The default adds 5 packages into the agent. If we create a single fixture we will need to parametrize all tests adding a lot more lines.

File: tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py (hunk 5 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py:96]: This is a fixture specific for this test, no reason to move it to conftest.py

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py:80]: Marking it on the test function allows for different tiers inside the same module. Using pytestmark is general. I would advise this method

File: tests/integration/test_vulnerability_detector/data/feeds/alas/custom_alas2_feed.json
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/data/feeds/alas/custom_alas2_feed.json:1]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/fe9ea3a83e949d0017641172b071c1d3ecc84377

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/cf428669ca6378ddac6a6dd7c45865851a1b1194

File: tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_cpe_helper/test_cpe_indexing_empty_fields.py]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/d2f3b68b5947526a287450a641ec45704024eac4

File: tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py (hunk 5 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/test_cpe_indexing.py]: done in https://github.com/wazuh/wazuh/pull/17576/commits/3749243c366e3a978305ce04345854fcd7d62079

File: tests/integration/test_vulnerability_detector/test_feeds/data/configuration_template/configuration_cpe_indexing.yaml (hunk 4 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_feeds/data/configuration_template/configuration_cpe_indexing.yaml]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/733738ed093a36accd09961bf60064388b0d78f3

File: .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
+name: Vulnerability Detector integration tests tier 0 and 1
+
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml:5]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/5ba6e4cb2a2c1d4388e117381b8575c8a30afad9

File: .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+          BRANCH_NAME=main
+          if [ "X`git ls-remote --heads https://github.com/wazuh/qa-integration-framework.git ${GITHUB_HEAD_REF}`" != "X" ]; then BRANCH_NAME=${GITHUB_HEAD_REF}; fi
+          git clone -b $BRANCH_NAME --single-branch https://github.com/wazuh/qa-integration-framework.git
+          sudo pip install qa-integration-framework/
+          sudo rm -rf qa-integration-framework/
+      # Run vulnerability detector integration tests.
+      - name: Run vulnerability detector integration tests
+        run: |
+          cd tests/integration
+          sudo python -m pytest --tier 0 --tier 1 test_vulnerability_detector/
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply .github/workflows/integration-tests-vulnerability_detector-tier-0-1.yml]: Done in https://github.com/wazuh/wazuh/pull/17576/commits/733738ed093a36accd09961bf60064388b0d78f3

**Comment by Deblintrake09** [comment]
Closing PR beacause branch is mixed with other commits through a merge. New PR will be opened, after branch is deleted and recreated.

**Pull request closed without merging**